code/modules/projectiles/guns/ego_gun/waw.dm

```diff
@@ -397,3 +397,24 @@
 		H.adjustBruteLoss(15)
 	..()
 	return
+
+//THIS IS ONLY A GUN WITH THE REVSYSTEM! IT HAS ONE TILE RANGE!!!
+/obj/item/gun/ego_gun/animalism
+	name = "animalism"
+	desc = "The frothing madness of the revving engine brings a fleeting warmth to your hands and heart alike."
+	icon_state = "animalism"
+	inhand_icon_state = "animalism"
+	ammo_type = /obj/item/ammo_casing/caseless/ego_animalism
+	weapon_weight = WEAPON_HEAVY
+	force = 40
+	damtype = RED_DAMAGE
+	armortype = RED_DAMAGE
+	spread = 0
+	attack_verb_continuous = list("saws", "tears", "lacerates", "cuts", "chops", "dices")
+	attack_verb_simple = list("saw", "tear", "lacerate", "cut", "chop", "dice")
+	hitsound = 'sound/weapons/chainsawhit.ogg'
+	autofire = 0.07 SECONDS
```

Comment:
I feel like this may cause lag easily. Perhaps converting to a non-projectile system? Instead checking in front of it when it fires?

Reply:
Not as easily, I would have to code a whole new autofire system from scratch.

Reply:
Changed it over to melee.